### ISSUE TITLE
Use "universal" GeckoView builds

### ISSUE THREAD

**Issue by pocmo** [mozilla-mobile/android-components#1447]
Not available yet, but hopefully soon:
https://bugzilla.mozilla.org/show_bug.cgi?id=1485045

With that we can:
* Get rid of all the build flavors for different architectures (yay!)
* Switch from `compileOnly` to `implementation` in our engine-gecko components.

**Comment by ncalexan** [comment]
An update on this work.  [Bug 1485045](https://bugzilla.mozilla.org/show_bug.cgi?id=1485045) was just preparatory work that @snorp landed a while back.  I just autolanded the first part of the work needed to _produce_ universal/multi-architecture fat AARs in TC in [Bug 1508976](https://bugzilla.mozilla.org/show_bug.cgi?id=1508976).  Follow-up ticket [Bug 1522581](https://bugzilla.mozilla.org/show_bug.cgi?id=1522581) tracks _publishing_ those fat AARs to `maven.mozilla.org`.  I am actively working on that piece now.

I have a patch against a-c that does "soup to nuts" (with Gradle composite builds thrown in, although that is blocked on several other tickets).  I see that a-c has an "abi" flavor dimension which ranges over armeabi-v7a, aarch64, x86, x86_64, universal.  Since all the GV stuff will need to ride trains (or, more likely, be uplifted) I propose to make `nightlyGeckoViewUniversal` do things and preserve the existing other variants.  When we have GV Nightly, Beta, _and_ Release published and working we can drop the non-universal variants, and eventually the "abi" flavor dimension entirely.  But this does need to ride the trains...

Sadly, there's additional coordination required for consumers: here's the [ticket for the Reference Browser](https://github.com/mozilla-mobile/reference-browser/issues/471).
